ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiInMemoryWorldStateKeyValueStorage.java

```diff
@@ -31,35 +31,53 @@ public BonsaiInMemoryWorldStateKeyValueStorage(
       final KeyValueStorage codeStorage,
       final KeyValueStorage storageStorage,
       final KeyValueStorage trieBranchStorage,
-      final KeyValueStorage trieLogStorage) {
-    super(accountStorage, codeStorage, storageStorage, trieBranchStorage, trieLogStorage);
+      final KeyValueStorage trieLogStorage,
+      final KeyValueStorage snapTrieBranchBucketStorage,
+      final KeyValueStorage snapTrieBranchSecondBucketStorage) {
+    super(
+        accountStorage,
+        codeStorage,
+        storageStorage,
```

Comment:
Can be contractStorageStorage but it seems to me no better. I think we will have to change this in another PR if we decide to do it because it will make even more changes to review